proposals/0293-extend-property-wrappers-to-function-and-closure-parameters.md

```diff
@@ -207,7 +207,7 @@ func buy(
 
 ## Detailed design
 
-Property wrappers are essentially sugar wrapping a given declaration with compiler-synthesized code. This proposal retains this principle. Annotating a parameter declaration with a property-wrapper attribute allows the call-site to pass a wrapped value or a projected value, and the compiler will automatically initialize the backing wrapper to pass to the function. The function author can also use the property-wrapper syntax for accessing the backing wrapper and the projected value in the body of the function.
+Property wrappers are essentially sugar wrapping a given declaration with compiler-synthesized code. Retaining this principle, a function can now be called with the wrapped and projected values. Namely, annotating a parameter declaration with a property-wrapper attribute changes the declaration’s type to the backing storage, and prompts the compiler to synthesize the wrapped and projected values. When the function is called the compiler will, also, insert a call to the appropriate property-wrapper initializer.
```

Comment:
I think these changes have obscured the meaning/purpose of this paragraph.

> Retaining this principle, a function can now be called with the wrapped and projected values

I don't think this makes sense without the context of "Annotating a parameter declaration with a property-wrapper attribute". Also, the fact that a function can be called with a wrapped value or a projected value of a property wrapper doesn't really follow from "retaining this principle"

> annotating a parameter declaration with a property-wrapper attribute changes the declaration’s type to the backing storage

This isn't something we want to state unless we're explicitly talking about how the code is compiled. As far as the user is concerned, the function type does _not_ take in the backing storage. That's why had worded the sentence to be about the call-site.

> and prompts the compiler to synthesize the wrapped and projected values

The wrapped and projected values are not synthesized, they're provided by the user.

I'm not sure why this last sentence was removed:

> The function author can also use the property-wrapper syntax for accessing the backing wrapper and the projected value in the body of the function

This is showcasing how the feature is useful for function authors because they now have property-wrapper syntax in the body of the function.

Reply:
Ok, I see your point. Upon rereading the paragraph, I realize that expecting to hear about the current design (since it's in the section's name), I read about existing behavior, which is then paralleled to new behavior. I find this problematic because the reader will –– up to this point –– be unfamiliar with this new behavior. Thus, what do you think of simply moving the first two periods in the end?

---

Annotating a parameter declaration with a property-wrapper attribute allows the call-site to pass a wrapped value or a projected value, and the compiler will automatically initialize the backing wrapper to be passed to the function. The function author can also use the property-wrapper syntax for accessing the backing wrapper and the projected value within the body of the function. **This behavior merely builds on property wrappers’ being sugar wrapping a given declaration with compiler-synthesized code.**

Reply:
That sounds good to me!